llvm/lib/IR/Operator.cpp

```diff
@@ -28,8 +28,9 @@ bool Operator::hasPoisonGeneratingFlags() const {
     return OBO->hasNoUnsignedWrap() || OBO->hasNoSignedWrap();
   }
   case Instruction::Trunc: {
-    auto *TI = dyn_cast<TruncInst>(this);
```

Comment:
This style is really bad. Either using a `cast` directly w/o a check or check the result before use.